core/src/block.rs

```diff
@@ -792,4 +792,58 @@ mod tests {
 
         Revalidate::revalidate::<true>(&valid_block, wsv).unwrap();
     }
+
+    #[test]
+    fn transactions_are_atomic() {
+        // Predefined world state
+        let alice_id = AccountId::from_str("alice@wonderland").expect("Valid");
+        let alice_keys = KeyPair::generate().expect("Valid");
+        let account =
+            Account::new(alice_id.clone(), [alice_keys.public_key().clone()]).build(&alice_id);
+        let domain_id = DomainId::from_str("wonderland").expect("Valid");
+        let mut domain = Domain::new(domain_id).build(&alice_id);
+        assert!(domain.add_account(account).is_none());
```

Comment:
Needs explanation.

Reply:
What exactly? it's standard think we use to create domain and accounts for wsv.